### PR TITLE
E2E test of issue #8785: Undefined when reading getUpdated

### PR DESCRIPTION
**Pull request by archibald1418**
Adding a test to reproduce the issue fixed in GH-8785

### Motivation and context

The issue dealt with inconsistent state of the user's job cache after he deletes a frame, tries to save the job, suddenly  gets an unexpected network error from the server, then tries to save the job again.

##### Arranged state (before() hook)
- Check deleted frame visibility box form the user's pop down menu (to confirm deletion later)
- Open a job from a task
- Pick a frame in the UI

##### Test case
- Delete the chosen frame
- Confirm that the deletion did happen (assert that a .restore element is visible)
- Try to save - but instead of 200 receive a stubbed response with 502
- Try to save again - this time yielding 200 OK without no unexpected errors

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new end-to-end test for job metadata consistency after frame deletion.
  
- **Bug Fixes**
	- Validated behavior when saving a job after frame deletion, ensuring proper error handling and recovery.

### CODE REVIEW

File: tests/cypress/e2e/issues_prs2/issue_8785_Cannot_read_properties_of_undefined_reading_getUpdated.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    const issueId = '8785';
+
+    function checkDeletedFrameVisibility() {
+        cy.openSettings();
+        cy.get('.cvat-workspace-settings-show-deleted').within(() => {
+            cy.get('[type="checkbox"]').should('not.be.checked').check();
+        });
+        cy.closeSettings();
+    }
+    function clickDelete() {
```
Comment (azhavoro) [inline tests/cypress/e2e/issues_prs2/issue_8785_Cannot_read_properties_of_undefined_reading_getUpdated.js:19]:
```suggestion
    function clickDeleteFrame() {
```

File: tests/cypress/e2e/issues_prs2/issue_8785_Cannot_read_properties_of_undefined_reading_getUpdated.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+            const routeMatcher = {
+                url: '/api/jobs/**/data/meta**',
+                method: 'PATCH',
+                times: 1, // cancels the intercept without retries
+            };
+
+            cy.intercept(routeMatcher, badResponse).as('patchError');
+
+            clickDelete();
```
Comment (azhavoro) [inline tests/cypress/e2e/issues_prs2/issue_8785_Cannot_read_properties_of_undefined_reading_getUpdated.js:53]:
```suggestion
            clickDeleteFrame();
```

File: tests/cypress/e2e/issues_prs2/issue_8785_update_job_metadata.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        cy.goToNextFrame(1);
+    });
+
+    describe(`Testing issue ${issueId}`, () => {
+        it('Crash on Save job. Save again.', () => {
+            const badResponse = { statusCode: 502, body: 'A horrible network error' };
+
+            cy.on('uncaught:exception', (err) => {
+                expect(err.code).to.equal(badResponse.statusCode);
+                expect(err.message).to.include(`> ${badResponse.body}`);
```
Comment (klakhov) [inline tests/cypress/e2e/issues_prs2/issue_8785_update_job_metadata.js:35]:
```suggestion
                expect(err.message).to.include(badResponse.body);
```
Lets not rely on symbols that we dont control. Checking body should be enough

**Review by klakhov — APPROVED** [review]
LGTM, dont forget to squash on merge :smile:

**Pull request merged**